components/bl-webcamPhoto-component/src/index.js

```diff
@@ -0,0 +1,60 @@
+import { useCallback, useMemo, useRef, useState } from 'react';
+
+import { MakePhotoButton, UploadButton } from './buttons/index';
+import { dataURLToBlob, toBase64 } from './helpers/file';
+import { checkMobile } from './helpers/device';
+import { Modal } from './modal';
+
+export default function WebcamPhoto({ component, eventHandlers }) {
+  const { uploadButtonLabel, makePhotoButtonLabel, buttonDisabled } = component;
+  const { onSaveImage } = eventHandlers;
+
+  const [modalVisibility, setModalVisibility] = useState(false);
+  const inputRef = useRef();
+
+  const handleClick = useCallback(() => {
+    setModalVisibility(true);
+  }, [modalVisibility]);
+
+  const handleChange = useCallback(async () => {
+    try {
+      const dataURL = await toBase64(inputRef.current.files[0]);
+
+      dataURLToBlob(dataURL)
+        .then(imageBlob => {
+          onSaveImage({ imageBlob });
+        });
+    } catch (error) {
+      console.error(`Webcam Photo: ${ error.message }`);
+    }
+  }, []);
+
+  const isMobile = useMemo(() => {
```

Comment:
we do not need to calculate the value for each component, the value won't be changed

Reply:
moved isMobile outside of the component, right?